src/sipsess/modify.c

```diff
@@ -161,8 +161,9 @@ int sipsess_reinvite(struct sipsess *sess, bool reset_ls)
  */
 int sipsess_modify(struct sipsess *sess, struct mbuf *desc)
 {
-	if (!sess || (!sess->established && !sess->refresh_allowed)
-	    || sess->terminated || sess->awaiting_answer)
+	if (!sess || (!sess->established && !sess->refresh_allowed &&
+	    mbuf_get_left(desc)) || sess->terminated || sess->awaiting_answer
+	    || !sip_dialog_established(sess->dlg))
```

Comment:
Please split up this into 2-3 different if-s

Reply:
Done.